### ISSUE TITLE
update c/c++ apis (6.0.0)

### ISSUE THREAD

**Issue by stweil** [tesseract-ocr/tesseract#4336]

**Comment by stweil** [comment]
Should be drop the C API in release 6? Or later? Who is still using it?

**Comment by egorpugin** [comment]
+1 for dropping.

There are some users most probably, but they can take `capi.h/cpp` into their projects or maintain separate thin library of these two files.

**Comment by amitdo** [comment]
Who is still using it?

https://github.com/nguyenq/tess4j/


Most likely it is used by many companies internally.

**Comment by stweil** [comment]
I know from other software that it's possible to mark API functions in the header files as "deprecated", so anybody who uses them in a build process gets a warning message. So we might start adding such deprecation markers and then hopefully get more feedback.

**Comment by amitdo** [comment]
Previous talk about the C API:

https://github.com/tesseract-ocr/tesseract/pull/4074#issuecomment-1546163064

**Comment by zdenop** [comment]
I still miss reasonable explanation of benefit for removing feature (C-API).
AFAIR contribution came from corporate environment for they want to use tesseract in pascal/Delphi(?). Also for new languages (e.g. rust) is easy to for with C library but problem with C++ (AFAIK).
If it is problem for somebody we can make it as optional feature (with default option to build it)

**Comment by egorpugin** [comment]
> AFAIR contribution came from corporate environment for they want to use tesseract in pascal/Delphi(?).

And they should maintain it.

We can mark capi as deprecated like @stweil said or unmaintained.

**Comment by jbarlow83** [comment]
I agree with removal. It's undocumented and in my limited experience, a bit of minefield for memory corruption bugs. pytesseract, tesseract-rs and ocrmypdf (my project) all use Tesseract in a subprocess. pytesseract used to have C API bindings but removed them many years ago.

To mark functions deprecated:
- `void myfunc() __attribute__ ((deprecated));` will mark a function as deprecated in GCC / LLVM.
- for MSVC ~~it's probably easiest to deprecate the whole compiler~~ okay I'll be nice, `__declspec(deprecated("** this is a deprecated function **"))`

**Comment by amitdo** [comment]
@zdenop,

Do you have an objection to move the C API out of the main tesseract repo, to a new repo 'tesseract-capI' ?

**Comment by stweil** [comment]
Would this be reasonable as long as we work with major version 5?

**Comment by egorpugin** [comment]
It is better to create v5 branch and do some fixes into both v5 and master. And declare master as future 'v6' branch.
